QEfficient/generation/text_generation_inference.py

```diff
@@ -61,6 +61,75 @@ def write_io_files(
         json.dump({"IO-files": io_files}, fp, indent=True)
 
 
+def get_compilation_batch_size(qpc_path: str):
+    qpc_base_path = os.path.dirname(qpc_path)
+    print(qpc_base_path)
+    specialization_file_path = os.path.join(qpc_base_path, "specializations.json")
+    print(specialization_file_path)
+    with open(specialization_file_path, "r") as file:
+        data = json.load(file)
+    compilation_batch_size = int(data["specializations"][0]["batch_size"])
+    return compilation_batch_size
+
+
+def check_batch_size_and_num_prompts(prompt: Union[str, List], compilation_batch_size: int):
+    if isinstance(prompt, list):
+        num_prompts = len(prompt)
+    elif isinstance(prompt, str):
+        num_prompts = 1
+    else:
+        print("Input prompt sould be either string for single input or List of string in case of mutliple inputs")
+    if compilation_batch_size > 1:
+        assert (
+            compilation_batch_size == num_prompts
+        ), f"Mismatch between number of prompts {num_prompts} and compilation batch size {compilation_batch_size}; please pass correct input argument"
+
+
+def read_prompts_txt_file(prompts_txt_file_path: str):
+    prompt = []
+    with open(prompts_txt_file_path, "r") as file:
+        for line in file:
+            prompt.append(line.strip())
+    return prompt
+
+
+def cloud_ai_100_exec_kv(
```

Comment:
we wanted to add `prompt_txt_file_path` as input to this Low level API right?

Reply:
updated